src/pages/LoginHandler/index.js

```diff
@@ -9,138 +9,89 @@ import {
   Title,
 } from '@patternfly/react-core';
 import AuthLayout from '@/components/AuthLayout';
-import { routerRedux } from 'dva/router';
 import { connect } from 'dva';
 import styles from './index.less';
 
-@connect(auth => ({
-  auth: auth.auth,
-}))
-class LoginHandler extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      username: '',
-      password: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    };
-  }
+const mapStateToProps = state => {
+  const { auth } = state;
+  return { auth };
+};
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+const LoginHandler = props => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [btnDisabled, setBtnDisabled] = useState(true);
 
-  setLoggedIn = payload => {
-    const { dispatch } = this.props;
+  /* eslint-disable camelcase */
+  const handleLoginSubmit = () => {
+    const { dispatch } = props;
     dispatch({
-      type: 'auth/loadUser',
-      payload,
-    });
-    dispatch(routerRedux.push(`/private`));
-  };
-
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
-  };
-
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
+      type: 'auth/loginUser',
+      payload: {
+        username,
```

Comment:
Are we relying on the server to validate emails as part of the login endpoint? We can perform validation on the client in order to reduce the number of bad requests making its way to the server and improve the load put on our API. We can make use of the `validateEmail` utility at `./utils/validator` here.

Reply:
Just fixed, thanks! @gurbirkalsi